File: RxTools-library/src/main/java/com/vondear/rxtools/view/dialog/RxDialogGPSCheck.java
package com.vondear.rxtools.view.dialog;

import android.app.Activity;
import android.content.Context;
import android.graphics.Color;
import android.view.View;

import com.vondear.rxtools.RxLocationTool;


/**
 * @author vondear
 * @date 2016/7/19
 * 检测Gps状态
 */
public class RxDialogGPSCheck extends RxDialogSureCancel {
    
    public RxDialogGPSCheck(Context context, int themeResId) {
        super(context, themeResId);
        initView();
    }

    public RxDialogGPSCheck(Context context, boolean cancelable, OnCancelListener cancelListener) {
        super(context, cancelable, cancelListener);
        initView();
    }

    public RxDialogGPSCheck(Context context) {
        super(context);
        initView();
    }

    public RxDialogGPSCheck(Activity context) {
        super(context);
        initView();
    }

    public RxDialogGPSCheck(Context context, float alpha, int gravity) {
        super(context, alpha, gravity);
        initView();
    }

    private void initView() {
        getTitleView().setBackgroundDrawable(null);
        setTitle("GPS未打开");
        getTitleView().setTextSize(16f);
        getTitleView().setTextColor(Color.BLACK);
        setContent("您需要在系统设置中打开GPS方可采集数据");
        getSureView().setText("去设置");
        getCancelView().setText("知道了");

        getSureView().setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                RxLocationTool.openGpsSettings(mContext);
                cancel();
            }
        });

        getCancelView().setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                cancel();
            }
        });
        setCanceledOnTouchOutside(false);
        setCancelable(false);
    }

}

File: RxUI/src/main/java/com/tamsiree/rxui/view/dialog/RxDialogGPSCheck.kt
package com.tamsiree.rxui.view.dialog

import android.app.Activity
import android.content.Context
import android.content.DialogInterface
import android.graphics.Color
import com.tamsiree.rxkit.RxLocationTool.openGpsSettings

/**
 * @author tamsiree
 * @date 2016/7/19
 * 检测Gps状态
 */
class RxDialogGPSCheck : RxDialogSureCancel {
    constructor(context: Context?, themeResId: Int) : super(context, themeResId) {
        initView()
    }

    constructor(context: Context?, cancelable: Boolean, cancelListener: DialogInterface.OnCancelListener?) : super(context, cancelable, cancelListener) {
        initView()
    }

    constructor(context: Context?) : super(context) {
        initView()
    }

    constructor(context: Activity?) : super(context) {
        initView()
    }

    constructor(context: Context?, alpha: Float, gravity: Int) : super(context, alpha, gravity) {
        initView()
    }

    private fun initView() {
        titleView.setBackgroundDrawable(null)
        setTitle("GPS未打开")
        titleView.textSize = 16f
        titleView.setTextColor(Color.BLACK)
        setContent("您需要在系统设置中打开GPS方可采集数据")
        sureView.text = "去设置"
        cancelView.text = "知道了"
        sureView.setOnClickListener {
            openGpsSettings(mContext)
            cancel()
        }
        cancelView.setOnClickListener { cancel() }
        setCanceledOnTouchOutside(false)
        setCancelable(false)
    }
}